ogr/ogrsf_frmts/miramon/ogrmiramondatasource.cpp

```diff
@@ -0,0 +1,310 @@
+    const char *pszExtension = CPLGetExtension(pszRootName);
+    char *pszFullMMLayerName;
+    if (EQUAL(pszExtension, "pol") || EQUAL(pszExtension, "arc") ||
+        EQUAL(pszExtension, "pnt"))
+    {
+        char *pszMMLayerName;
+        pszMMLayerName = CPLStrdup(CPLResetExtension(pszRootName, ""));
+        pszMMLayerName[strlen(pszMMLayerName) - 1] = '\0';
+
+        pszFullMMLayerName = CPLStrdup((const char *)pszMMLayerName);
+
+        // Checking that the folder where to write exists
+        const char *szDestFolder = CPLGetDirname(pszFullMMLayerName);
+        char **papszDirContent = nullptr;
+        papszDirContent = VSIReadDir(szDestFolder);
+        if (!papszDirContent)
+        {
+            CPLFree(pszMMLayerName);
+            CPLFree(pszFullMMLayerName);
+            CPLError(CE_Failure, CPLE_AppDefined,
+                     "The folder %s does not exist.", szDestFolder);
+            return nullptr;
+        }
+        else
+            CSLDestroy(papszDirContent);
+        CPLFree(pszMMLayerName);
+    }
+    else
+    {
+        const char *osPath;
+
+        osPath = pszRootName;
+        pszFullMMLayerName =
+            CPLStrdup(CPLFormFilename(pszRootName, pszLayerName, ""));
+
+        /* -------------------------------------------------------------------- */
+        /*      Let's create the folder if it's not already created.            */
+        /*      (only the las level of the folder)                              */
+        /* -------------------------------------------------------------------- */
+        char **papszDirContent = nullptr;
+        papszDirContent = VSIReadDir(osPath);
+        if (!papszDirContent)
+        {
+            if (VSIMkdir(osPath, 0755) != 0)
+            {
+                CPLFree(pszFullMMLayerName);
+                CPLError(CE_Failure, CPLE_AppDefined,
+                         "Unable to create the folder %s.", pszRootName);
+                return nullptr;
+            }
+        }
+        else
+            CSLDestroy(papszDirContent);
+    }
+
+    /* -------------------------------------------------------------------- */
+    /*      Return open layer handle.                                       */
+    /* -------------------------------------------------------------------- */
+    if (Open(pszFullMMLayerName, nullptr, poSRS, TRUE, papszOptions))
+    {
+        CPLFree(pszFullMMLayerName);
+        auto poLayer = papoLayers[nLayers - 1];
+        return poLayer;
+    }
+
+    CPLFree(pszFullMMLayerName);
+    return nullptr;
+}
+
+/****************************************************************************/
+/*                           TestCapability()                               */
+/****************************************************************************/
+
+int OGRMiraMonDataSource::TestCapability(const char *pszCap)
+
+{
+    if (EQUAL(pszCap, ODsCCreateLayer))
+        return TRUE;
```

Comment:
```suggestion
        return bUpdate;
```

Reply:
A mistery to me :)

Reply:
What is mysterious ? If code queries for the CreateLayer capability, you answer true if and only if the dataset has been opened in update mode (which is only triggered if a new dataset is generated, as updating an existing dataset is not supported by the Miramon driver)